src/lib/datepicker/calendar.ts

```diff
@@ -29,7 +30,131 @@ import {MatDatepickerIntl} from './datepicker-intl';
 import {MatMonthView} from './month-view';
 import {MatMultiYearView, yearsPerPage} from './multi-year-view';
 import {MatYearView} from './year-view';
+import {ComponentPortal, ComponentType, Portal} from '@angular/cdk/portal';
+import {Subject} from 'rxjs/Subject';
+import {takeUntil} from 'rxjs/operators/takeUntil';
 
+/** Default header for MatCalendar */
+@Component({
+  moduleId: module.id,
+  selector: 'mat-calendar-header',
+  templateUrl: 'calendar-header.html',
+  encapsulation: ViewEncapsulation.None,
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MatCalendarHeader<D> implements OnDestroy {
+  /** Subject that emits when the component has been destroyed. */
+  private _destroyed = new Subject<void>();
+
+  constructor(private _intl: MatDatepickerIntl,
+              @Host() public calendar: MatCalendar<D>,
```

Comment:
@tobiasschweizer I'm seeing a number of failures when trying to presubmit this related to injecting the calendar here. I think you may need to do this:

```ts
@Host() @Inject(forwardRef(() => MatCalendar)) public calendar: MatCalendar<D>
```

Since `MatCalendarHeader` is declared before `MatCalendar`

Reply:
@mmalerba Good point, which test gave you the error?

fixed in https://github.com/angular/material2/pull/9639/commits/822dd563fc3d334e994708eb2fc90adb0088071d

Reply:
Tests in Google's codebase for other project teams. I can't link to them since they're internal, I'll rerun the presubmit with the fix though.

In the mean time, it looks like you need to resolve some conflicts again.

Reply:
ok, merge done in https://github.com/angular/material2/pull/9639/commits/ba4e839275847f8582c423fd64da002a9e3d1549